core/app/models/spree/line_item.rb

```diff
@@ -138,49 +114,68 @@ def options=(options={})
       opts = options.dup # we will be deleting from the hash, so leave the caller's copy intact
 
       currency = opts.delete(:currency) || order.try(:currency)
+      set_price(currency, opts)
+
+      self.assign_attributes opts
+    end
 
+    private
+    # Sets the quantity to zero if it is nil or less than zero.
+    def invalid_quantity_check
```

Comment:
If we already renaming ann moving stuff (what I like), we should rename this method into `normalize_quantity`, because that's what it does.

PS: And please add a new line after the `private`. My dear eyes are bleeding.